### DISCUSSION TITLE
Relation managers - Filter select options

### DISCUSSION THREAD

**Discussion by icaliman** [filamentphp/filament#4278] [Help]
I have two models: **Company** and **Department**. 

A **Company** has many **Departments**. 

And a **Department** has a parent **Department**.

I have a form for creating company departments:
<img width="933" alt="image" src="https://user-images.githubusercontent.com/4498933/193070030-29f9ef88-9189-4f53-9ce2-61329157f802.png">

How to display only the company departments in the **Parent** select?

**Comment by icaliman** [comment]
I found a solution by using **Multi-tenancy** 

https://filamentphp.com/docs/2.x/admin/resources/getting-started#multi-tenancy

**Reply by sergioraldana** [reply]
Hello, can you explain more deep how to achieve that?

**Comment by cobraprojects** [comment]
for `Select` and `MultiSelect` you can pass the relation and a custom query if needed
```
Select::make('authorId')
    ->relationship('author', 'name', fn (Builder $query) => $query->withTrashed())
```

you can also create your custom select options
```
Select::make('authorId')
    ->options(fn (Model $record) =>  $record->relation_or_query->pluck('name', 'id'))
```
